src/native/managed/cdacreader/src/Contracts/Metadata.cs

```diff
@@ -0,0 +1,386 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Reflection.Metadata.Ecma335;
+using UntrustedMethodTable = Microsoft.Diagnostics.DataContractReader.Contracts.UntrustedMethodTable_1;
+using MethodTable = Microsoft.Diagnostics.DataContractReader.Contracts.MethodTable_1;
```

Comment:
I'm trying something here that I'm not completely sure about - the Metadata contract needs to create and consume these MethodTable values, but it's quite likely that their internals would be different once the Metadata contract evolves.  But initially it's annoying to write `MethodTable_1` everywhere.

An alternative is to make a completely opaque `MethodTableHandle` that forms the contract's interface and it will likely carry a pointer back to the contract and the contract will have some kind of dictionary to retrieve the actual interesting method table data.

Reply:
I decided I don't like this.  I'm going to make `MethodTable` completely opaque for the public contract surface and put all the query methods on `Metadata`